docs/source/data/fetching-rest.mdx

```diff
@@ -0,0 +1,423 @@
+
+## HTTP Methods
+
+`RESTDataSource` includes convenience methods for common REST API request methods: `get`, `post`, `put`, `patch`, and `delete` ([see the source](https://github.com/apollographql/datasource-rest/blob/25862e18d8b35e324c150654c5686ed317b3fca8/src/RESTDataSource.ts#L163)).
+
+An example of each is shown below:
+
+<ExpansionPanel title="Click to expand">
+
+```ts
+class MoviesAPI extends RESTDataSource {
+  override baseURL = 'https://movies-api.example.com/';
+
+  // GET
+  async getMovie(id) {
+    return this.get(
+      `movies/${encodeURIComponent(id)}`, // path
+    );
+  }
+
+  // POST
+  async postMovie(movie) {
+    return this.post(
+      `movies`, // path
+      movie, // request body
+    );
+  }
+
+  // PUT
+  async newMovie(movie) {
+    return this.put(
+      `movies`, // path
+      { body: { movie } }, // request body
+    );
+  }
+
+  // PATCH
+  async updateMovie(movie) {
+    return this.patch(
+      `movies`, // path
+      { body: { id: movie.id, movie } }, // request body
+    );
+  }
+
+  // DELETE
+  async deleteMovie(movie) {
+    return this.delete(
+      `movies/${encodeURIComponent(movie.id)}`, // path
+    );
+  }
+}
+
+```
+
+</ExpansionPanel>
+
+Note the use of [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) in the above snippet. This is a standard function that encodes special characters in a URI, preventing a possible injection attack vector.
+
+For a simple example, suppose our REST endpoint responded to the following URLs:
+
+- DELETE `/movies/:id`
+- DELETE `/movies/:id/characters`
+
+A "malicious" client could provide an `:id` of `1/characters` to target the delete `characters` endpoint when it was the singular `movie` endpoint that we were trying to delete. URI encoding prevents this kind of injection by transforming the `/` into `%2F`. This can then be correctly decoded and interpreted by the server and won't be treated as a path segment.
+
+### Method parameters
+
+For all HTTP convenience methods, the **first parameter** is the relative path of the endpoint you're sending the request to (e.g., `movies`).
+
+The **second parameter** depends on the HTTP method:
+
+- For HTTP methods with a request body (`post`, `put`, `patch`), the second parameter is an _object_ that contains the `method` `cacheOptions`, and request `body` (e.g., `{ body: { id: movie.id } }`).
+- For HTTP methods _without_ a request body, the second parameter is an object with keys and values corresponding to the request's query parameters.
+
+## Intercepting fetches
+
+> **New in Apollo Server 4**: Apollo Server 4 now uses [`@apollo/fetcher`](../migration#apolloutilsfetcher-replaces-apollo-server-env) under-the-hood for fetching, so hooks like `willSendRequest` no longer have direct access to `Request` or `Headers` methods.
```

Comment:
```suggestion
> **New in Apollo Server 4**: Apollo Server 4 now uses the [`@apollo/utils.fetcher`](../migration#apolloutilsfetcher-replaces-apollo-server-env) interface under the hood for fetching. This interface lets you choose your own implementation of the Fetch API. To ensure compatibility with all Fetch implementations, the request provided to hooks like `willSendRequest` is a plain JS object rather than a `Request` object with methods.
```

maybe something like this?